src/providers/database.ts

```diff
@@ -307,6 +307,8 @@ export class RefBuilder {
   };
 }
 
+const resourceRegex = /^projects\/([^/]+)\/instances\/([a-zA-Z0-9-]+?)\/refs(\/.+)?/;
```

Comment:
I'm not convinced (yet!) that we need non-greedy operator on the capturing group for instance based on the provided test cases:

e.g.
```js
const greedy = /^projects\/([^/]+)\/instances\/([a-zA-Z0-9-]+)\/refs(\/.+)?/
const nongreedy = /^projects\/([^/]+)\/instances\/([a-zA-Z0-9-]+?)\/refs(\/.+)?/

console.log("projects/_/instances/foo/refs/root/refs".match(greedy)[2]) //  => foo
console.log("projects/_/instances/foo/refs/root/refs".match(nongreedy)[2]) // => foo

console.log("projects/_/instances/foo/refs/refs".match(greedy)[2])  // => foo
console.log("projects/_/instances/foo/refs/refs".match(nongreedy)[2]) // => foo
```

Can you share edge cases that require non-greedy operator?

Reply:
Yeh I think you're right, now we've taken out the `/` from the set in the instances capture group the non-greediness is no longer needed - shall update 👍